### PR TITLE
MNT Lint the codebase

### PR DESCRIPTION
**Pull request by adrinjalali**
This PR adds linter pre-commit hooks and runs them on the codebase.

It also changes the line length to 88 which is the default for `black`, since it seems most libraries are adopting this.

We can then add the hash of this commit to the ignored ones so that `git blame` doesn't complain.

This is to move our codebase to a more consistent state, and so that we can add a linter to the CI and avoid running CI if the liter fails.

cc @fairlearn/fairlearn-maintainers

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+repos:
```
Comment (riedgar-ms) [inline .pre-commit-config.yaml:1]: I'm guessing that this is the critical file, with the rest being `black` doing its worst?

File: pyproject.toml [hunk 2]
```diff
@@ -1,5 +1,10 @@
 [tool.black]
-line-length = 79
+line-length = 88
```
Comment (riedgar-ms) [inline pyproject.toml:2]: Ahh, this too?

**Comment by riedgar-ms** [comment]
I've no objections in principle to this, but please can you highlight which files you've changed, since most of the changes are simply `black` being let off its leash.

**Comment by adrinjalali** [comment]
Not sure why we get the circular import errors here, haven't changed any imports, I'll try to find it.

**Comment by adrinjalali** [comment]
@riedgar-ms CI is green :)

**Review by riedgar-ms — APPROVED** [review]
I'm happy with this; I doubt there will ever be a "good" time to do this, so we may as well go with "now."

**Review by MiroDudik — APPROVED** [review]

**Pull request merged**